src/main/java/mate.academy/lib/Injector.java

```diff
@@ -8,6 +20,45 @@ public static Injector getInjector() {
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Object clazzImplementationInstance = createNewInstance(clazz);
+        for (Field field : clazz.getDeclaredFields()) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = createNewInstance(findImplementation(field.getType()));
+                field.setAccessible(true);
+                try {
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (clazz.isAnnotationPresent(Component.class)) {
+            try {
+                Constructor<?> constructor = clazz.getConstructor();
+                return constructor.newInstance();
+            } catch (NoSuchMethodException | InvocationTargetException
+                     | InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException("Can`t create new instance of " + clazz.getName());
```

Comment:
Replace the multiple exceptions with ReflectiveOperationException to handle all reflection-related exceptions. Also, include the cause of the exception in the RuntimeException.